### ISSUE TITLE
Missing user-understandable error message with GOOGLE_CREDENTIALS being unset

### ISSUE THREAD

**Issue by sttts** [openshift/installer#3044]
Following the 4.2 OpenShift install docs I tried to create a GCP cluster, without having GOOGLE_CREDENTIALS set.

I ended up with:

```
$ ./openshift-install --dir gcp create cluster --log-level=debug
...
? Platform gcp
INFO Credentials loaded from gcloud CLI defaults
? Project ID
? Region europe-west3
DEBUG       Generating Base Domain...
FATAL failed to fetch Terraform Variables: failed to fetch dependency of "Terraform Variables": failed to fetch dependency of "Cluster ID": failed to fetch dependency of "Install Config": failed to generate asset "Base Domain": could not retrieve base domains: Get https://www.googleapis.com/dns/v1/projects//managedZones?alt=json&prettyPrint=false: context canceled
```

This is not understandable by a user. It wasn't for me.

**Comment by sttts** [comment]
After I found out how to set GOOGLE_CREDENTIALS, I hit another error where the user has no clue about the problem:

```
FATAL failed to fetch Terraform Variables: failed to fetch dependency of "Terraform Variables": failed to generate asset "Image": failed to fetch RHCOS metadata: open data/rhcos-amd64.json: no such file or directory
```

**Comment by patrickdillon** [comment]
The problem that is being identified in the first comment is that we accept an invalid `Project`, which then causes the base domain prompt to fail.

**Comment by patrickdillon** [comment]
/close
resolved by #3484

**Comment by sttts** [comment]
Thank you!
